src/main/java/org/quiltmc/loader/impl/game/MappingConfigurationImpl.java

```diff
@@ -200,9 +204,34 @@ private void initialize() {
 		}
 
 		this.namespaces = new ArrayList<>();
-		namespaces.add(mappings.getSrcNamespace());
+		if (mappings.getSrcNamespace() != null) {
+			namespaces.add(mappings.getSrcNamespace());
+		}
 		namespaces.addAll(mappings.getDstNamespaces());
 		this.namespaces = Collections.unmodifiableList(namespaces);
+
+		if (!namespaces.isEmpty()) {
+			Log.info(LogCategory.MAPPINGS, "Loaded mapping namespaces: %s", namespaces);
+		} else {
+			Log.warn(LogCategory.MAPPINGS, "No mappings namespaces were loaded!");
+			mappings = null;
+		}
+
+		Log.info(LogCategory.MAPPINGS, "Target namespace: %s", getTargetNamespace());
+
+		if (!namespaces.contains(getTargetNamespace())) {
+			if (!namespaces.isEmpty()) {
+				throw new IllegalStateException(String.format("Requested target namespace %s not loaded. Available options: %s", targetNamespace, namespaces));
+			} else {
+				if (getTargetNamespace().equals("official")) {
+					Log.warn(LogCategory.MAPPINGS, "Continuing without mappings because target namespace is 'official'");
+				} else {
+					throw new IllegalStateException("Requested target namespace %s not loaded. To continue without mappings, " +
```

Comment:
You either need to use string concatenation here or an explicit `String.format` to handle the `%s` as a target namespace
```suggestion
					throw new IllegalStateException("Requested target namespace " + targetNamespace + " not loaded. To continue without mappings, " +
```